tldw_Server_API/app/core/RAG/rag_service/unified_pipeline.py

```
@@ -5236,28 +5239,36 @@ async def _retrieve_for_claim(c_text: str, top_k: int = 5):
+                                    sqlite3.Error,
+                                ),
+                            ) as db:
+                                # Collect media IDs present in documents
+                                media_ids: list[int] = []
+                                for d in result.documents:
+                                    try:
+                                        mid = d.metadata.get("media_id") if isinstance(d.metadata, dict) else None
+                                        if mid is not None:
+                                            media_ids.append(int(mid))
+                                    except (TypeError, ValueError):
+                                        continue
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)